bundle-workflow/src/test_workflow/test_publisher.py

```diff
@@ -0,0 +1,39 @@
+import os
+import sys
+
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), "../../src"))
+from aws.s3_bucket import S3Bucket
+
+
+class TestPublisher:
```

Comment:
I suggest renaming this to `TestResults`, since it wraps test results, and rename `publish_test_results_to_s3` to `to_s3`, similar to other `to_file` or `to_dict` that we have on other classes.

I would add `TestRecorder.results` that returns an instance of test results.

Remove `s3_bucket` from the constructor since it never consumes S3 results, and make it a parameter in `to_s3`, so `TestResults().to_s3(bucket)`.

Reply:
I will change the name in the proposal as well https://github.com/opensearch-project/opensearch-build/issues/207#issuecomment-911927824

Reply:
`TestPublisher` feels right to me as this class really just publishes the results while wrapping them up in the right directory structure. But I would leave it to dB who is our master of renaming :).

Reply:
@dblock any thoughts on this?

Reply:
No strong feelings.